### DISCUSSION TITLE
Difference between trafficRouting.managedRoutes and trafficRouting.istio.virtualService.routes ?

### DISCUSSION THREAD

**Discussion by bgreen9595** [argoproj/argo-rollouts#3722] [Q&A]
In a canary deployment with Istio configuration/subset splitting, what is the difference between the routes specified below? The documentation specs don't give a clear distinction, other than the managedRoutes are required for setHeaderRoute and setMirrorRoute

trafficRouting:
   managedRoutes:
        - name: route-1
        - name: route-2
   istio:
      virtualService:
         routes:
         - route-1
         - route-2

**Comment by kostis-codefresh** [comment]
I am not an expert on istio but I believe that the second syntax is the "generic" one for all traffic providers. The first one is istio specific and as the documentation says it is needed only for header based routing.

I believe that if you use the first syntax, the second one is implied and not really needed.

If not, please open a github issue. 

Alternatively just using both syntaxes at the same time, shouldn't be a problem.
